Source/Scene/PostProcess.js

```diff
@@ -0,0 +1,570 @@
+     *     '    gl_FragColor = vec4(color.rgb * scale + offset, 1.0);\n' +
+     *     '}\n';
+     * scene.postProcessCollection.add(new Cesium.PostProcess({
+     *     fragmentShader : fs,
+     *     uniformValues : {
+     *         scale : 1.1,
+     *         offset : function() {
+     *             return new Cesium.Cartesian3(0.1, 0.2, 0.3);
+     *         }
+     *     }
+     * }));
+     */
+    function PostProcess(options) {
+        var fragmentShader = options.fragmentShader;
+        var textureScale = defaultValue(options.textureScale, 1.0);
+        var pixelFormat = defaultValue(options.pixelFormat, PixelFormat.RGBA);
+
+        //>>includeStart('debug', pragmas.debug);
+        Check.typeOf.object('options', options);
+        Check.typeOf.string('options.fragmentShader', fragmentShader);
+        Check.typeOf.number.greaterThan('options.textureScale', textureScale, 0.0);
+        Check.typeOf.number.lessThanOrEquals('options.textureScale', textureScale, 1.0);
+        if (!PixelFormat.isColorFormat(pixelFormat)) {
+            throw new DeveloperError('options.pixelFormat must be a color format.');
+        }
+        //>>includeEnd('debug');
+
+        this._fragmentShader = fragmentShader;
+        this._uniformValues = options.uniformValues;
+        this._textureScale = textureScale;
+        this._forcePowerOfTwo = defaultValue(options.forcePowerOfTwo, false);
+        this._sampleMode = defaultValue(options.samplingMode, PostProcessSampleMode.NEAREST);
+        this._pixelFormat = pixelFormat;
+        this._pixelDatatype = defaultValue(options.pixelDatatype, PixelDatatype.UNSIGNED_BYTE);
+        this._clearColor = defaultValue(options.clearColor, Color.BLACK);
+
+        this._uniformMap = undefined;
+        this._command = undefined;
+
+        this._colorTexture = undefined;
+        this._depthTexture = undefined;
+
+        this._actualUniformValues = {};
+        this._dirtyUniforms = [];
+        this._texturesToRelease = [];
+        this._texturesToCreate = [];
+        this._texturePromise = undefined;
+
+        this._passState = new PassState();
+        this._passState.scissorTest = {
+            enabled : true,
+            rectangle : defined(options.scissorRectangle) ? BoundingRectangle.clone(options.scissorRectangle) : new BoundingRectangle()
+        };
+
+        this._ready = true;
+
+        this._name = options.name;
+        if (!defined(this._name)) {
+            this._name = createGuid();
+        }
+
+        // set by PostProcessCollection
+        this._collection = undefined;
+        this._index = undefined;
+
+        /**
+         * Whether or not to execute this post-process when ready.
+         *
+         * @type {Boolean}
+         */
+        this.enabled = true;
+        this._enabled = this.enabled;
+    }
+
+    defineProperties(PostProcess.prototype, {
+        /**
+         * Determines if this post-process is ready to be executed.
```

Comment:
Perhaps a bit more context - why might it not be ready?  Because it is waiting on textures to load?